### PR TITLE
EVG-13650 Add option to change RDP password for windows hosts

### PR DESCRIPTION
**Pull request by khelif96**
Backend PR https://github.com/evergreen-ci/evergreen/pull/4291
Commit https://github.com/evergreen-ci/spruce/commit/3270a162b3f0f3e5c75b9920570e3d92d858344d Is temporary to fix some types that broke when the underlying schema changed in a non backwards compatable way before the corresponding front end changes were merged. It will be removed after a more permanent change is merged. Lets talk about it tomorrow. cc. @malikchaya2  @SupaJoon 

Windows Hosts
image

Non Windows Hosts 
image

### CODE REVIEW

File: src/gql/queries/get-task.ts [hunk 1]
```diff
@@ -69,6 +69,7 @@ export const GET_TASK = gql`
       annotation {
         taskId
         taskExecution
+        userCanModify
```
Comment (malikchaya2) [inline src/gql/queries/get-task.ts:72]: Why is this needed here if you don't use this value anywhere in your PR?

File: src/pages/spawn/spawnHost/EditSpawnHostModal.tsx [hunk 2]
```diff
@@ -215,6 +221,45 @@ export const EditSpawnHostModal: React.FC<EditSpawnHostModalProps> = ({
...
+                      servicePassword: e.target.value,
+                    })
+                  }
+                  id="rdpPasswordInput"
+                  type="password"
+                />
+                <Tooltip
+                  title={
+                    <>
+                      Password should match the criteria defined{" "}
```
Comment (malikchaya2) [inline src/pages/spawn/spawnHost/EditSpawnHostModal.tsx:247]: nice!

File: src/gql/mutations/edit-spawn-host.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ export const EDIT_SPAWN_HOST = gql`
     $instanceType: String
     $expiration: Time
     $noExpiration: Boolean
+    $servicePassword: String
```
Comment (malikchaya2) [inline src/gql/mutations/edit-spawn-host.ts:13]: Is there a corresponding backend PR for this?

File: src/gql/queries/get-task.ts (hunk 1 above)
Comment (khelif96) [reply src/gql/queries/get-task.ts:72]: One of the components in spruce depends on having this type. So I had to include it to appease typescript

**Review by malikchaya2 — APPROVED** [review]

File: src/gql/queries/get-task.ts (hunk 1 above)
Comment (malikchaya2) [reply src/gql/queries/get-task.ts:72]: So you remember which component?

File: src/gql/mutations/edit-spawn-host.ts (hunk 3 above)
Comment (khelif96) [reply src/gql/mutations/edit-spawn-host.ts:13]: https://github.com/evergreen-ci/evergreen/pull/4291

**Review by SupaJoon — APPROVED** [review]
